### ISSUE TITLE
✨ Generalise Construction of Interaction Graph

### ISSUE THREAD

**Issue by ystade** [cda-tum/mqt-core#599]
### What's the problem this feature will solve?

The list of supported gates for the construction of an interaction graph (see [src/datastructures/Layer.cpp#L155](https://github.com/cda-tum/mqt-core/blob/df425d332e51f3e07a29089b04be337f773920a4/src/datastructures/Layer.cpp#L155)) is rather limited at the moment. I see no reason why this shouldn't support a broader gateset. This does not properly handle two-target gates yet (specifically, `SWAP, iSWAP, iSWAPdg, Peres, Peresdg, DCX, ECR, RXX, RYY, RZZ, RZX, XXminusYY, XXplusYY`) and also does not handle CompoundOperations (although that could be transparently handled by the caller).

_Originally posted by @burgholzer in https://github.com/cda-tum/mqt-core/pull/596#discussion_r1584808824_

### Describe the solution you'd like

The mentioned method should be capable of handling most (all?) possible Gates.

**Comment by ystade** [comment]
@burgholzer Yes, we can do that. I mean, NALAC can still be installed via an old version of MQT QMAP that should be around for a while still.

**Comment by burgholzer** [comment]
Yeah 👍🏼 definitely. I would also not backport the respective PR to the `v3.x` branch so that it stays around there as well.
I'll submit a PR to remove the respective code.

**Comment by burgholzer** [comment]
@ystade potentially a stupid question, but are any of the `datastructures` classes still used by code other than NALAC?
Or could the entire module be removed? I am not seeing any non-test uses in MQT Core.
